### PR TITLE
#2 - Rewrite Registration

### PR DESCRIPTION
**Pull request by ImmortalIdiot**
I have realized two-step registration form (next, "signup screen"). The signup screen consists of two parts: before clicking the primary button and after clicking the primary button. In the first part of signup event, the user enters student id, email (he/she need email to reset password) and password. If the user entered this data correctly and clicked on the primary button, then the user will be redirected to the second part of the signup screen. Here the user must enter their full name and group and click on the button. If all the data is correct, the user will be register in database (there is no possibility of registration in the database yet).

### CODE REVIEW

File: app/src/main/res/values/strings.xml
Comment (karpetfishensky) [inline app/src/main/res/values/strings.xml:1]: А зачем строка пустая между app_name и signup?

File: app/src/main/java/com/immortalidiot/rutlead/models/SignUpModel.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+data class SignUpModel(
+    val email: String,
+    val studentID: String,
+    val password: String,
+    val group: String,
+    val name: String,
+    val isFocused: Boolean,
+    val isPasswordVisible: Boolean
+)
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/models/SignUpModel.kt:11]:
```suggestion
)

```

File: app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt [hunk 2]
```diff
@@ -104,20 +105,22 @@ fun LoginDesign(
...
                 .background(color = palette.primary)
                 .border(
                     width = dimensions.borderSSmall,
                     color = palette.outline,
                     shape = roundedShape
                 )
+                .padding(
+                    top = dimensions.verticalBigPadding,
+                    bottom = dimensions.verticalBigPadding
+                )
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:118]:
```suggestion
                .padding(vertical = dimensions.verticalBigPadding)
```

File: app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt [hunk 3]
```diff
@@ -166,15 +169,17 @@ fun LoginDesign(
...
+                PrimaryButton(
+                    modifier = modifier
+                        .fillMaxHeight(0.14f)
+                        .fillMaxWidth(0.55f),
                     palette = palette,
                     text = "Войти",
                     onButtonClick = {
                         focusManager.clearFocus()
                         keyboardController?.hide()
                         viewModel.request()
```
Comment (LittlePony00) [inline app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:181]:
```suggestion
                    onButtonClick = remember {
                    {
                        focusManager.clearFocus()
                        keyboardController?.hide()
                        viewModel.request()
                    }
```

Comment (LittlePony00) [reply app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:181]: https://habr.com/ru/companies/ozontech/articles/742854/#2.4

File: app/src/main/res/values/strings.xml
Comment (ImmortalIdiot) [reply app/src/main/res/values/strings.xml:1]: У меня нет пустой строки, кинь скрин.

File: app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt (hunk 2 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:118]: Oh, thanks. I forgot that this parameter exists

File: app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt (hunk 3 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt:184]: idk why this statement doesn't work

File: app/src/main/java/com/immortalidiot/rutlead/models/SignUpModel.kt (hunk 1 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/models/SignUpModel.kt]: Done

File: app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt (hunk 2 above)
Comment (ImmortalIdiot) [reply app/src/main/java/com/immortalidiot/rutlead/screens/LoginScreen.kt]: Done

**Review by LittlePony00 — APPROVED** [review]

**Pull request merged**